better packaging for pyston_lite

kmod opened the pull request:
This is on top of #218

I wrote setup.py files for the main extension as well as pyston_lite_autoload which automatically enables pyston_lite. I'm open to the idea of just having a single extension that is always enabled.

I went ahead and uploaded some preliminary versions to pypi:

```
$ python3.8 -m venv /tmp/env
$ time /tmp/env/bin/python -c 'for i in range(100000000): pass'
2.89s
$ /tmp/env/bin/pip install pyston_lite_autoload
$ time /tmp/env/bin/python -c 'for i in range(100000000): pass'
2.36s
```

kmod committed:
Simple setup.py

not sure yet how to run the preprocessing steps through setup.py though

also includes a wheel-building step via manylinux

kmod committed:
Package the autoloader package

I also changed it to default-on instead of default-off

undingen reviewed (approved):
Cool I just tried pip installing `pyston_lite_autoload` into a fedora docker and it seemed to work fine :)

kmod committed:
bump version numbers from the dev version

and pin autoload to the same version of pyston_lite

not 100% sure that's necessary, but if we're going to encourage people to do
`pip install pyston_lite_autoload`, it'd be weird to get a different version of pyston_lite

kmod committed:
Improvements to building pyston-lite

- do the custom preprocessing steps inside the setup.py
- clean up before and after the build

undingen commented:
Just took a quick look at your changes, they look good to me. Not sure whats the current state of this but feel free to merge it if it's ready and you are just waiting for me to merge it.

kmod commented:
The CI kept failing so I was investigating, but it looks like it's passing now :)
